pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -230,6 +231,11 @@ func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMoun
 	if osdProps.metadataDevice != "" {
 		envVars = append(envVars, metadataDeviceEnvVar(osdProps.metadataDevice))
 	}
+	if extraEnvVarsValue, err := k8sutil.GetOperatorSetting(context.TODO(), c.context.Clientset, controller.OperatorSettingConfigMapName, "OSD_EXTRA_ENV_VARS", ""); err != nil {
```

Comment:
Why does the prepare job need this? I thought the args were needed only to start the OSD process.

Reply:
Correct, so the `--mkfs` also needs this? If so, let's add a small comment in the code why the prepare stage needs it too :)